payment/service.go

```diff
@@ -750,92 +750,80 @@ func (s *Service) GetCredentialSigningKeys(ctx context.Context, merchantID strin
 	return resp, nil
 }
 
-var (
-	errInvalidIssuanceInterval = errors.New("invalid issuance interval")
-	errOrderExpired            = errors.New("order is expired")
-)
-
 // credChunkFn - given a time, calculate the next increment of time based on interval
-func credChunkFn(interval timeutils.ISODuration) func(time.Time, bool) time.Time {
-	return func(t time.Time, next bool) time.Time {
-		var result time.Time
+func credChunkFn(interval timeutils.ISODuration) func(time.Time) (time.Time, time.Time) {
+	return func(t time.Time) (time.Time, time.Time) {
+		var (
+			start time.Time
+			end   time.Time
+		)
 
 		// get the future time one credential interval away
 		c, err := interval.From(t)
 		if err != nil {
-			return time.Time{}
+			return start, end
 		}
 		// get the go duration to that future time one credential away
 		td := (*c).Sub(t)
 		// truncate that time based on the duration it is away
 		// i.e. 1 day will truncate on the day
 		// i.e. 1 month will truncate on the month
-		if next {
-			t = t.Add(td)
-		}
+		end = t.Add(td)
 
 		switch interval.String() {
 		case "P1M":
 			y, m, _ := t.Date()
 			// reset the date to be the first of the given month
-			result = time.Date(y, m, 1, 0, 0, 0, 0, time.UTC)
+			start = time.Date(y, m, 1, 0, 0, 0, 0, time.UTC)
+
+			y1, m1, _ := end.Date()
+			end = time.Date(y1, m1, 1, 0, 0, 0, 0, time.UTC)
 		default:
 			// use truncate
-			result = t.Truncate(td)
+			start = t.Truncate(td)
+			end = end.Truncate(td)
 		}
-		return result
+		return start, end
```

Comment:
I think we should eventually move this into a `ISODuration.Truncate` method - can you open an issue?